uaclient/tests/test_ua_timer.py

```diff
@@ -2,117 +2,122 @@
-    run_jobs,
-    UPDATE_MESSAGING_INTERVAL,
-    UPDATE_STATUS_INTERVAL,
-    GCP_AUTO_ATTACH_INTERVAL,
-)
-from uaclient.jobs.update_messaging import update_apt_and_motd_messages
-from uaclient.jobs.update_state import update_status
-from uaclient.jobs.gcp_auto_attach import gcp_auto_attach
 
-
-class TestRunJob:
+class TestTimedJob:
     @pytest.mark.parametrize("caplog_text", [logging.DEBUG], indirect=True)
     def test_run_job_returns_true_on_successful_job_run(
         self, FakeConfig, caplog_text
     ):
         """Return True on successful job run."""
-        cfg = FakeConfig
+        cfg = FakeConfig()
 
         def success_job(config):
             assert config == cfg
 
-        assert True is run_job(
-            job_name="Day Job", job_func=success_job, cfg=cfg
-        )
-        assert "Running job: Day Job" in caplog_text()
+        job = TimedJob("day_job", success_job, 14400)
+        assert True is job.run(cfg)
+        assert "Running job: day_job" in caplog_text()
 
     @pytest.mark.parametrize("caplog_text", [logging.WARNING], indirect=True)
     def test_run_job_returns_false_on_failed_job(
         self, FakeConfig, caplog_text
     ):
         """Return False on failed job run and warns in log."""
-        cfg = FakeConfig
+        cfg = FakeConfig()
 
         def failed_job(config):
             assert config == cfg
             raise Exception("Something broke")
 
-        assert False is run_job(
-            job_name="Day Job", job_func=failed_job, cfg=cfg
-        )
-        assert "Error executing job Day Job: Something broke" in caplog_text()
+        job = TimedJob("day_job", failed_job, 14400)
+        assert False is job.run(cfg)
+        assert "Error executing job day_job: Something broke" in caplog_text()
+
+    @pytest.mark.parametrize("is_cfg_set", (False, True))
+    def test_get_default_run_interval(self, is_cfg_set, FakeConfig):
+        """Use the default run interval when config is absent or invalid."""
+        cfg = FakeConfig()
+        if is_cfg_set:
+            setattr(cfg, "day_job_timer", None)
+        job = TimedJob("day_job", lambda: None, 14400)
+
+        assert 14400 == job.run_interval_seconds(cfg)
+
+    def test_get_configured_run_interval(self, FakeConfig):
+        """Use the configured run interval when not overriden."""
+        cfg = FakeConfig()
+        setattr(cfg, "day_job_timer", 28800)
+        job = TimedJob("day_job", lambda: None, 14400)
+
+        assert 28800 == job.run_interval_seconds(cfg)
+
+    @pytest.mark.parametrize(
+        "disabled_by_default,interval_value", ((False, 14400), (True, 0))
+    )
+    @pytest.mark.parametrize("caplog_text", [logging.DEBUG], indirect=True)
+    def test_does_not_run_if_disabled(
+        self, FakeConfig, caplog_text, disabled_by_default, interval_value
+    ):
+        cfg = FakeConfig()
+        if not disabled_by_default:
+            setattr(cfg, "day_job_timer", 0)
+
+        def disabled_job(config):
```

Comment:
I think we can turn this into a mock and assert that the mock was never called, just to be sure that the reason the call failed is because of the timer and not the function itself